packages/components/pie-modal/src/index.ts

```diff
@@ -360,18 +360,23 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
      * @private
      */
     private renderModalContentAndFooter (): TemplateResult {
+        const hasActions = this.leadingAction || this.supportingAction;
```

Comment:
this will be a truthy value if consumers pass an empty object {}. 
considering leadingAction.text is a required property, I would check for it to ensure the action isn't empty 
```suggestion
        const hasActions = this.leadingAction?.text || this.supportingAction?.text;
```